### PR TITLE
Add logic to convert from old scheduling format to new one - #3

### PR DESCRIPTION
**Pull request by GoZaddy**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] Check logging
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
+package org.folio.de.entity.bursarlegacy;
```
Comment (ncovercash) [inline src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java:1]: Should this have landed in git? Or is it generated?

File: src/main/java/org/folio/des/controller/FolioTenantController.java [hunk 2]
```diff
@@ -60,4 +108,314 @@ public ResponseEntity<Void> postTenant(TenantAttributes tenantAttributes) {
     return tenantInit;
   }
 
+  private boolean isLegacyJob(LegacyJob job) {
```
Comment (ncovercash) [inline src/main/java/org/folio/des/controller/FolioTenantController.java]: Can we extract some of the legacy job logic into a separate method somewhere?

**Review by ncovercash — APPROVED** [review]

**Review by ncovercash — CHANGES_REQUESTED** [review]
oops hit wrong button

File: src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java (hunk 1 above)
Comment (GoZaddy) [reply src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java:1]: Yeahh, it's how basically how JPA maps the database rows to Java objects. There's [a similar one](https://github.com/ualibweb/mod-data-export-spring/blob/master/src/main/java/org/folio/de/entity/Job.java) for the new `Job` as well.

Comment (ncovercash) [reply src/main/java/org/folio/de/entity/bursarlegacy/LegacyJob.java:1]: Ah right; just wanted to make sure we weren't duplicating generated code! LGTM

File: src/main/java/org/folio/des/controller/FolioTenantController.java (hunk 2 above)
Comment (GoZaddy) [reply src/main/java/org/folio/des/controller/FolioTenantController.java]: @ncovercash Yeah, it did feel out of place but I wasn't sure where to put it. What do you think about moving it to a `LegacyBursarUtil` class in the `util` package?

Comment (ncovercash) [reply src/main/java/org/folio/des/controller/FolioTenantController.java]: `util` package sounds great, I'd recommend maybe something like `LegacyBursarMigrationUtil` for clarity tho

Comment (GoZaddy) [reply src/main/java/org/folio/des/controller/FolioTenantController.java]: Just took care of that!

File: src/main/java/org/folio/des/controller/FolioTenantController.java [hunk 3]
```diff
@@ -60,4 +109,299 @@ public ResponseEntity<Void> postTenant(TenantAttributes tenantAttributes) {
     return tenantInit;
   }
 
+  public void recreateLegacyJobs() {
```
Comment (ncovercash) [inline src/main/java/org/folio/des/controller/FolioTenantController.java]: @GoZaddy could we move this in there too?

**Review by ncovercash — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/des/controller/FolioTenantController.java (hunk 3 above)
Comment (GoZaddy) [reply src/main/java/org/folio/des/controller/FolioTenantController.java]: For sure!

Comment (GoZaddy) [reply src/main/java/org/folio/des/controller/FolioTenantController.java]: @ncovercash done!

**Review by ncovercash — APPROVED** [review]
LGTM! Thanks for the hard work on this — now go study for calc 3, lol

**Comment by GoZaddy** [comment]
Lolll, I definitely need to

**Pull request merged**